nilearn/decoding/tests/test_decoder.py

```diff
@@ -187,6 +188,58 @@ def test_check_parameter_grid_is_empty(rand_X_Y):
     assert param_grid == {}
 
 
+@pytest.mark.parametrize(
+    "param_grid, need_wrap",
+    [
+        ({"alphas": [1, 10, 100, 1000]}, True),
+        ({"alphas": [1, 10, 100, 1000], "fit_intercept": [True, False]}, True),
+        ({"fit_intercept": [True, False]}, False),
+        ({"alphas": [[1, 10, 100, 1000]]}, False),
+        ({"alphas": (1, 10, 100, 1000)}, True),
+        ({"alphas": [(1, 10, 100, 1000)]}, False),
+        ({"alphas": ((1, 10, 100, 1000),)}, False),
+        ({"alphas": np.array([1, 10, 100, 1000])}, True),
+        ({"alphas": [np.array([1, 10, 100, 1000])]}, False),
+        ([{"alphas": [1, 10]}, {"alphas": [[100, 1000]]}], True),
+        ([{"alphas": [1, 10]}, {"fit_intercept": [True, False]}], True),
+        (None, False),
+    ],
+)
+def test_wrap_param_grid(param_grid, need_wrap):
```

Comment:
I think this tests too many things.

I would split this into:
- one test that checks that the warning is thrown (not sure it is worth checking the warning is not thrown)
- one test for  `param_grid_wrapped = _wrap_param_grid(None, param_name)`
- one test for the "is param_name in param_grid_wrapped?"

This should remove of lot of the "logic" you need to go through when putting everything in a single test